pd/src/testnet.rs

```diff
@@ -0,0 +1,671 @@
+use std::env::current_dir;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::{fmt, fs::File};
+
+use anyhow::Result;
+use directories::UserDirs;
+use regex::{Captures, Regex};
+use serde::{de, Deserialize};
+
+use penumbra_crypto::Address;
+use tendermint::PrivateKey;
+
+use crate::genesis;
+
+/// Methods and types used for generating testnet configurations.
+
+pub fn parse_allocations_file(input_file: PathBuf) -> Result<Vec<TestnetAllocation>> {
+    let file = match File::open(&input_file) {
+        Err(why) => panic!("couldn't open {}: {}", input_file.display(), why),
```

Comment:
Instead of a `panic!`, I would recommend using the `.context()` method from the `anyhow::Context` trait to add additional context to the error, then propagating it as an `anyhow::Error`.